src/fuzzy.rs

```diff
@@ -11,7 +11,7 @@ const WILDCARD: &str = "...";
 
 /// Does `s` conform to the fuzzy pattern `pattern`? Note that `plines` is expected not to start or
 /// end with blank lines, and each line is expected to be `trim`ed.
-pub(crate) fn match_vec(plines: &[&str], s: &str) -> bool {
+pub(crate) fn match_vec(plines: &Vec<String>, s: &str) -> bool {
```

Comment:
I'm confused here. Why do we need to pass a reference to the `Vec` instead? As far as I can tell, `match_vec` is only ever called on the same thread as it's caller, so you wouldn't need to use an `Arc` if you were passing a reference.

Reply:
This should be `&[String]` -- I'll fix that. But I think your question is really "why `String` and not `&str`?" I needed that in an intermediate transformation but I'm no longer sure if I need it. Let me try going back.

Reply:
Okay

Reply:
It is no longer needed, which is nice! The new commit fixes this, moving us back to `&str` and reducing copying.